parquet/src/main/java/org/apache/iceberg/parquet/ParquetReader.java

```diff
@@ -120,18 +125,27 @@ public boolean hasNext() {
 
     @Override
     public T next() {
-      if (valuesRead >= nextRowGroupStart) {
-        advance();
-      }
-
-      if (reuseContainers) {
-        this.last = model.read(last);
-      } else {
-        this.last = model.read(null);
+      try {
+        if (valuesRead >= nextRowGroupStart) {
+          advance();
+        }
+
+        if (reuseContainers) {
+          this.last = model.read(last);
+        } else {
+          this.last = model.read(null);
+        }
+        valuesRead += 1;
+
+        return last;
+      } catch (ParquetDecodingException e) {
+        if (reader != null) {
+          // Knowing the exact parquet file is essential for tracing bad nodes
+          // that produced the corrupt file, parquet lib doesn't do this today.
+          LOG.error("Error decoding Parquet file {}", reader.getFile(), e);
```

Comment:
ParquetInputFile doesn't implement `toString` method, right? Does this log print human-readable path? 

```java
  public String getFile() {
    return file.toString();
  }
```

Reply:
Could we just print the `location`?

Reply:
`location` is a method on the `org.apache.iceberg.io.OutputFile` interface, and not `org.apache.iceberg.io.InputFile`.